# Perkenalan

Saya I Wayan Budi(H8 FTDS-001), seorang Data Scientist dari PT Sangkuters Antam Battery akan melakukan perhitungan kondisi baterai 

# LiveCode1 Phase0

Problems

Kamu adalah seorang data scientist yang bekerja di salah satu perusahaan manufaktur baterai Lithium ion. Tim produksi meminta kamu untuk mengecek apakah baterai yang mereka buat dalam kondisi bagus atau tidak. Sebelum melakukan pengecekan lebih lanjut, kamu diminta untuk menghitung hambatan/resistansi dan Daya setiap detik:

[R=\frac{V}{I}]

[P=VI]

    R adalah Kapasitas dalam satuan Ohm
    P adalah Daya dalam Watt
    I adalah kuat arus dalam satuan Ampere
    V adalah tegangan dalam satuan Volt

Hitung kapasitas baterai tiap waktunya untuk cycle ganjil dan daya baterai tiap waktunya untuk cycle genap dari dataset dengan ketentuan sebagai berikut:

    Gunakan function untuk menghitung hambatan dan daya
    Harus menerapkan konsep looping, conditional if, dan pandas query
Dataset

Dataset dapat diakses pada link berikut: https://github.com/fahmimnalfrzki/Dataset/blob/main/NMC1-9.xlsx?raw=true

Keterangan kolom:

    time(s): Detik ke- pengukuran
    V_m: Voltage/tegangan yang terukur tiap waktu
    Current (A): Kuat arus terukur tiap waktu
    Cycle: Nomor siklus charge-discharge baterai
    Cap: Kapasitas baterai tiap waktu


# Import

In [6]:
import pandas as pd #mengimpor pandas

# Data Loading

In [7]:
df= pd.read_excel('NMC1-9.xlsx') #membaca excel

<bound method NDFrame.head of        time(s)    V_m  Current (A)  Cycle       Cap
0            0  4.098        0.997      1  2.840817
1            1  4.096        0.998      1  2.840817
2            2  4.094        0.999      1  2.840817
3            3  4.093        0.998      1  2.840817
4            4  4.092        0.999      1  2.840817
...        ...    ...          ...    ...       ...
90977    90977  2.506        1.002      9  2.770180
90978    90978  2.504        1.002      9  2.770180
90979    90979  2.502        1.002      9  2.770180
90980    90980  2.500        1.002      9  2.770180
90981    90981  2.497        1.002      9  2.770180

[90982 rows x 5 columns]>

# Answers

In [ ]:
Membuat Function & Pandas Query

In [20]:
df.info()#query Pandas
def hambatan(): #menghitung hambatan
    df.loc[index,"Hambatan3"] = df.loc[index,"V_m"] / df.loc[index,"Current (A)"]
    return df

def daya():#menghitung daya
    df.loc[index, "Daya3"] = df.loc[index, "V_m"] * df.loc[index, "Current (A)"]
    return df


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90982 entries, 0 to 90981
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   time(s)      90982 non-null  int64  
 1   V_m          90982 non-null  float64
 2   Current (A)  90982 non-null  float64
 3   Cycle        90982 non-null  int64  
 4   Cap          90982 non-null  float64
 5   Daya3        50540 non-null  float64
 6   Hambatan3    40442 non-null  float64
dtypes: float64(5), int64(2)
memory usage: 4.9 MB


Terlihat jika 90982 baris data, dan 5 kolom dari data, sedangkan daya3 dan hambatan adalah tambahan

In [ ]:
Conditional If dan Loop, Pandas Query

In [11]:
for index, row in df.iterrows():
    if df.loc[(index, "Cycle")] %2 ==0:
        df.loc[index, "Daya3"] = df.loc[index, "V_m"] * df.loc[index, "Current (A)"]
    else:
        df.loc[index,"Hambatan3"] = df.loc[index,"V_m"] / df.loc[index,"Current (A)"] #kalkulasi 
df.head()

time(s)    V_m  Current (A)  Cycle       Cap     Daya3  Hambatan3
0        0  4.098        0.997      1  2.840817  4.085706        NaN
1        1  4.096        0.998      1  2.840817  4.087808        NaN
2        2  4.094        0.999      1  2.840817  4.089906        NaN
3        3  4.093        0.998      1  2.840817  4.084814        NaN
4        4  4.092        0.999      1  2.840817  4.087908        NaN

In [ ]:
Mohon maaf functionnya & hasil terakhirnya tidak dimasukkan karena menjalankan Query sangat lama